src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/DeserializationHelpers.cs

```diff
@@ -57,6 +58,37 @@ internal static class DeserializationHelpers
             }
         }
 
+
+        internal static void ApplyInstanceAnnotations(object resource, IEdmStructuredTypeReference structuredType, ODataResourceWrapper resourceWrapper, 
+            ODataDeserializerProvider deserializerProvider, ODataDeserializerContext readContext)
+        {
+            IDelta delta = resource as IDelta;
+            
+            PropertyInfo propertyInfo = EdmLibHelpers.GetInstanceAnnotationsDictionary(structuredType.StructuredDefinition(), readContext.Model);
+            if (propertyInfo == null)
+            {
+                return;
+            }
+
+            object value;
+            if (delta != null)
+            {                
+                delta.TryGetPropertyValue(propertyInfo.Name, out value);
+            }
+            else
+            {
+                value = propertyInfo.GetValue(resource);
+            }
+
+            Dictionary<string, IDictionary<string, object>> instAnnotationDictionary = value as Dictionary<string, IDictionary<string, object>>;
+            if (instAnnotationDictionary == null)
```

Comment:
>instAnnotationDictionary [](start = 16, length = 24)

there are three different cases in which instAnnotationDictionary could be null:
1) the propertyInfo returned in line 67 was not defined on the type
2) the type of the propertyInfo returned in line 67 was not able to be cast to Dictionary<string,IDictionary<string,object>>
3) the propertyInfo was a property defined on the type that could be cast to Dictionary<string,IDictionary<string,object>>, but whose value was null.

I think you're trying to handle the 3rd case, and assuming that the first 2 are already handled?

Reply:
Yes, The propertyinfo is got from PropertyInfo propertyInfo = EdmLibHelpers.GetInstanceAnnotationsDictionary(structuredType.StructuredDefinition(), readContext.Model);
which only gets typeof InstanceAnnotation property (IDictionary<string,IDictionary<string,object>>) ,so the check of existence of such a property and the defined type is done there